react/ProductContextProvider.js

```diff
@@ -47,25 +49,26 @@ class ProductContextProvider extends Component {
   }
 
   render() {
-    const {
-      data,
-      params: { slug },
-    } = this.props
-    const { loading, product } = data
-    const { categories } = product || {}
+    const { data, params: { slug }, client } = this.props
+    const { loading } = data
+    const productPreview = client.readFragment({
+      id: buildCacheLocator('vtex.store-graphql@2.x', 'Product', slug),
+      fragment: productPreviewFragment
+    })
+    const product = loading ? productPreview : data.product
 
     const productQuery = {
       product,
-      loading,
+      loading
     }
 
     return (
       <div className="vtex-product-details-container">
         <Fragment>
-          {!loading && <MicroData product={product} />}
+          {product && <MicroData product={product} />}
           {React.cloneElement(this.props.children, {
             productQuery,
-            categories,
+            categories: product && product.categories,
```

Comment:
We were going to kill this prop. Its more simple send `productQuery` to the children.